### PR TITLE
Ja traj clustering

### PR DESCRIPTION
**Pull request by joacolongo**
Trajectory clustering protocol added

### CODE REVIEW

File: pwchem/protocols/MolecularDynamics/protocol_import_MDSystem.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                      condition='outputType == 2')
+        form.addParam('inputTopology', params.PathParam, label='Topology file',
+                       help='Topology file - Formats can be PDB, TOP, TPR, PRMTOP, PARM7, CMS and XML')
+        form.addParam('addTraj', params.BooleanParam, default=False, label='Trajectory file')
+        form.addParam('inputTrajectory', params.PathParam, label='Trajectory file', condition='addTraj',
+                       help='Trajectory file - Formats can be XTC, NC, NETCDF and DCD')
+
+        form.addParam('hasLig', params.BooleanParam, label='Has ligand?', default=False,
+                       help='Does the system has a ligand?')
+        form.addParam('ligID', params.StringParam, label='Has ligand?', condition='hasLig',
```
Comment (DaniDelHoyo) [inline pwchem/protocols/MolecularDynamics/protocol_import_MDSystem.py]: Cambia el label, es igual que el parametro anterior

File: pwchem/protocols/MolecularDynamics/protocol_import_MDSystem.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                       help='Does the system has a ligand?', default='LIG')
+
+    # --------------------------- INSERT steps functions --------------------
+    def _insertAllSteps(self):
+        self._insertFunctionStep('importStep')
```
Comment (DaniDelHoyo) [inline pwchem/protocols/MolecularDynamics/protocol_import_MDSystem.py]: aqui la nueva tendencia de Scipion es pasarle no el nombre de la funcion en str sino la funcion tal cual:
'importStep' -> self.importStep

(sin comillas y con el self)

File: pwchem/protocols/MolecularDynamics/protocol_trajectory_clustering.py [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+        mode = int(self.clusterMode.get())
+
+        args = '-f "{}" '.format(trajFile)
+        args += '-t "{}" '.format(topFile)
+        args += '-s {} '.format(self.stride.get())
+        args += '-l clustering.log '
+        args += '-sa "{}" '.format(alignStr)
+        args += '-sr "{}" '.format(rmsdStr)
+        args += '-m {} '.format(CLUSTERING_METHODS[int(self.clusterMethod.get())])
+        # args += '-axis default -limitmat 100000000 -i Y '
```
Comment (DaniDelHoyo) [inline pwchem/protocols/MolecularDynamics/protocol_trajectory_clustering.py]: Checkea estos mensajes de sonarcloud. Algunos son tonterias, pero por tener el codigo limpito

File: pwchem/scripts/mdtraj_PCA.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+    args = parser.parse_args()
+
+    topo = mdtraj.load(args.inputFilename)
+    traj = mdtraj.load(args.inputTraj, top=topo)
+
+    if args.pca_coord:
+        plotPcCoords(traj)
+
+    elif args.pca_dist:
```
Comment (DaniDelHoyo) [inline pwchem/scripts/mdtraj_PCA.py]: solo se plotea 1 u otro? no se pueden los 2 en la misma llamada como en mdtraj_SS?
si es que no, ignora el comentario

File: pwchem/scripts/trajectory_clustering.py [hunk 5]
```diff
@@ -0,0 +1,919 @@
...
+    Determine the optimal number of clusters using KMeans and the elbow method.
+    Returns the optimal k (int).
+    """
+    from sklearn.cluster import KMeans
+    from scipy.spatial.distance import cdist
+
+    distortions = []
+    K = range(2, 15)
+    for k in K:
+        kMeans = KMeans(n_clusters=k, n_init=10)
```
Comment (DaniDelHoyo) [inline pwchem/scripts/trajectory_clustering.py]: como dice sonarcloud, estaría bien poder setear la semilla en estas cosas aleatorias (probablemente haya mas sitios que se podría hacer, si ves alguno, añadelo tambien)
Añadelo como un parametro avanzado en en protocolo que llame esto, con el valor de tu numero favorito por ejemplo

File: pwchem/tests/tests_molecular_dynamics.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    def test_prolif(self):
+        self._runProLIF()
+        assertHandle(self.assertIsNotNone,getattr(self.prolifSystem, 'outputSystem', None))
+        assertHandle(
+            self.assertIsNotNone,
+            hasattr(self.prolifSystem, '_prolifFile'),
+            cwd=self.prolifSystem.getWorkingDir()
+        )
+
+class TestTrajClustering(BaseTest):
```
Comment (DaniDelHoyo) [inline pwchem/tests/tests_molecular_dynamics.py]: con vistas a que probablemente se añadan mas tests que usen este import de MDSystem, igual se puede hacer un MDBaseTest o algo asi que tenga ya la funcion para no duplicar en cada test que lo tenga que hacer

File: pwchem/__init__.py [hunk 7]
```diff
@@ -169,7 +169,6 @@ def addOpenbabelPackage(cls, env, default=True):
                                       f'openbabel={OPENBABEL_DIC["version"]} pymol-open-source')\
             .addCondaPackages(['swig', 'plip', 'pdbfixer'], channel='conda-forge')\
             .addCondaPackages(['clustalo', 'pip=25'], channel='bioconda', targetName='CLUSTALO_INSTALLED') \
-            .addCondaPackages(['mdanalysis'], channel='conda-forge',  targetName='MDANALYSIS_INSTALLED') \
```
Comment (DaniDelHoyo) [inline pwchem/__init__.py:172]: no usamos ya MDAnalysis para nada? comprueba bien que no lo use ningun otro viewer ni nada pls

**Review by DaniDelHoyo — CHANGES_REQUESTED** [review]
En general el codigo está genial, apenas te pediría cambios, pero chekea los pocos comentarios que te he hecho y alguno de SonarCloud tambien para que quede el codigo limpio

File: pwchem/__init__.py (hunk 7 above)
Comment (joacolongo) [reply pwchem/__init__.py:172]: Lo añadí a el entorno de MDtraj que me pareció que cuadraba más

File: pwchem/objects/base.py [hunk 8]
```diff
@@ -1775,8 +1775,9 @@ def __init__(self, filename=None, **kwargs):
     self._ff = pwobj.String(kwargs.get('ff', None))
     self._wff = pwobj.String(kwargs.get('wff', None))
 
-    self._ligName = pwobj.String(kwargs.get('ligName', 'LIG'))
+    self._ligName = pwobj.String(kwargs.get('ligName', None))
```
Comment (DaniDelHoyo) [inline pwchem/objects/base.py]: Creo recordar que el LIG como nombre por defecto era algo que se utilizaba en algun momento... estás seguro de que no rompe nada dejarlo en None?

**Review by DaniDelHoyo — CHANGES_REQUESTED** [review]

**Review by DaniDelHoyo — APPROVED** [review]
Good to go
